### PR TITLE
Handle announcement only streams.

### LINKED ISSUE #2711: Sending message to an admin-send-only stream doesn't fail properly.
bug1
Here, #announce is a stream where only admins may post. Instead of trying to send the message, the app should tell the user it's not possible.
Here is how the server handles this:
image

### PR DESCRIPTION
**Pull request by roberthoenig**
Fixes #2711.

### CODE REVIEW

**Comment by roberthoenig** [comment]
Whoops, forgot to post the demo gifs.

### Normal user
bug7

### Admin user
bug8

File: src/message/AnnouncementOnly.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <Label
+          style={styles.text}
+          text="Only organization admins are allowed to post to this stream."
+        />
+      </View>
+    );
+  }
+}
+
+export default connect()(AnnouncementOnly);
```
Comment (borisyankov) [inline src/message/AnnouncementOnly.js]: You don't need `connect`.

File: src/streams/streamSelectors.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* @flow */
+import { createSelector } from 'reselect';
+
+import type { Narrow } from '../types';
+import { getStreams } from '../directSelectors';
+import { isStreamOrTopicNarrow } from '../utils/narrow';
+
+export const getIsActiveStreamAnnouncementOnly = (narrow: Narrow) =>
```
[the remark this answers is not in the document]
Comment (borisyankov) [reply src/streams/streamSelectors.js]: Feel free to create a new commit for that in this PR.

**Comment by roberthoenig** [comment]
Updated:
* Rebased
* Fixed the FloxFixMe
* Removed the unnecessary connect()
* moved getIsActiveStreamAnnouncementOnly to subscriptionSelectors and avoided creating a separate streamSelectors. I won't do the refactor of extracting the selectors in subscriptionSelectors that are actually stream selectors here, but can do that in another PR.

**Comment by roberthoenig** [comment]
Rebased.

**Comment by roberthoenig** [comment]
Updated:

* added the i18n string
* extracted the `is_announcement_only` flow type into a new commit
* aligned the import statements in ComposeBox.*.js
* Swapped `styles: Extract disabledCompose* styles.` and `Handle narrow to announcement only stream.`

**Comment by gnprice** [comment]
Merged -- thanks @roberthoenig !

The revisions look great; pulling the corrected type into its own commit was a particularly good idea.

A couple of small issues which I fixed before merging:
* That first commit correcting the type actually caused type errors on its own -- it needed to go together with the updates in `nullObjects.js`. (I used the same technique Tim demonstrated last week to pull those changes forward into the first commit.)
* The last commit message still explained why we showed this message in preference to the "not subscribed" message when both applied; it hadn't been updated for the revised UI choice.

**Pull request merged**